### PR TITLE
Add Conan package manager usage details and simplify README page

### PR DESCRIPTION
**Pull request by ihsandemir**
*Added Conan package manager usage details.

*Restructured the README to be simpler and shorter and carried the detailed documentation to `Reference_Manual.md` file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3200 +1,18 @@
...
-
-# 9. Getting Help
-
-You can use the following channels for your questions and development/usage issues:
-
-* This repository by opening an issue.
-* Hazelcast Hazelcast channel on Gitter: 
-[![Join the chat at https://gitter.im/hazelcast/hazelcast](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/hazelcast/hazelcast?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
-* Our Google Groups directory: https://groups.google.com/forum/#!forum/hazelcast
+* [Hazelcast Community Slack](https://slack.hazelcast.com/)
```
Comment (yuce) [inline README.md:8]: Would it be better to link to the `#cpp-client` channel?

File: README.md [hunk 2]
```diff
@@ -1,3200 +1,18 @@
...
+* Hazelcast Hazelcast channel on [Gitter](https://gitter.im/hazelcast/hazelcast)
```
Comment (Serdaro) [inline README.md:9]: I don't know if it is intentional, but still: "Hazelcast" is duplicated here.

Comment (yuce) [inline README.md:9]: Is one of the `Hazelcast`s superfluous?

**Review by yuce — CHANGES_REQUESTED** [review]
LGTM, except 2 trivial issues.

**Review by Serdaro — COMMENTED** [review]
Minor comment.

File: README.md (hunk 1 above)
Comment (ihsandemir) [reply README.md:8]: Will this link work? `https://slack.hazelcast.com/#cpp-client`

File: README.md (hunk 2 above)
Comment (ihsandemir) [reply README.md]: fixed.

Comment (ihsandemir) [reply README.md]: fixed.

**Review by Serdaro — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (yuce) [reply README.md:8]: This one seems to work: https://hazelcastcommunity.slack.com/channels/cpp-client

File: README.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ C++11 support. Detailed requirements can be found at [Requirements](Reference_Ma
 The library can be installed using package managers ([Conan](Reference_Manual.md#113-conan-users)) or directly [from source code](Reference_Manual.md#112-install-from-source-code-using-cmake) using [CMake](https://cmake.org/).
 
 * [Hazelcast Community Slack](https://slack.hazelcast.com/)
-* Hazelcast Hazelcast channel on [Gitter](https://gitter.im/hazelcast/hazelcast)
+* Hazelcast channel on [Gitter](https://gitter.im/hazelcast/hazelcast)
   [![Join the chat at https://gitter.im/hazelcast/hazelcast](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/hazelcast/hazelcast?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
```
Comment (burakcelebi) [inline README.md:10]: I think it is OK to remove gitter now. It's deprecated and replaced with Slack.

File: README.md [hunk 4]
```diff
@@ -1,3200 +1,18 @@
 # Async Hazelcast C++ Client
-# Table of Contents
```
Comment (burakcelebi) [inline README.md:2]: I would start with a desc of Hazelcast:

> Hazelcast is an open-source distributed in-memory data store and computation platform. It provides a wide variety of distributed data structures and concurrency primitives.

File: README.md (hunk 1 above)
Comment (ihsandemir) [reply README.md]: changed.

File: README.md (hunk 3 above)
Comment (ihsandemir) [reply README.md]: Removed.

File: README.md (hunk 4 above)
Comment (ihsandemir) [reply README.md:2]: Added.

**Review by yuce — DISMISSED** [review]
Thanks, LGTM

File: Reference_Manual.md [hunk 5]
```diff
@@ -0,0 +1,3223 @@
...
+
+* [Introduction](#introduction)
+* [Resources](#resources)
+* [Release Notes](#release-notes)
+* [1. Getting Started](#1-getting-started)
+  * [1.1. Installing](#11-installing)
+    * [1.1.1. Requirements](#111-requirements)
+    * [1.1.2. Install From Source Code Using CMake](#112-install-from-source-code-using-cmake)
+      * [1.1.2.1. Downloading Source Code](#1121-downloading-source-code)
+      * [1.1.2.2 Linux and MacOS Users](#1122-linux-and-macos-users)
```
Comment (yemreinci) [inline Reference_Manual.md:11]:
```suggestion
      * [1.1.2.2. Linux and MacOS Users](#1122-linux-and-macos-users)
```

File: Reference_Manual.md [hunk 6]
```diff
@@ -0,0 +1,3223 @@
...
+      * [1.1.2.3. Windows Users](#1123-windows-users)
+      * [1.1.2.4 Advanced Installation](#1124-advanced-installation)
```
Comment (yemreinci) [inline Reference_Manual.md:13]:
```suggestion
      * [1.1.2.4. Advanced Installation](#1124-advanced-installation)
```

File: Reference_Manual.md [hunk 7]
```diff
@@ -0,0 +1,3223 @@
+# Hazelcast C++ Client Library 
+
+* [Introduction](#introduction)
+* [Resources](#resources)
+* [Release Notes](#release-notes)
+* [1. Getting Started](#1-getting-started)
+  * [1.1. Installing](#11-installing)
+    * [1.1.1. Requirements](#111-requirements)
```
Comment (yemreinci) [inline Reference_Manual.md:8]: Better to put these requirements under the "installing from source" section. Package managers will enforce/install the requirements themselves.

**Review by yemreinci — CHANGES_REQUESTED** [review]

File: Reference_Manual.md (hunk 7 above)
Comment (ihsandemir) [reply Reference_Manual.md:8]: Changed.

**Review by yemreinci — APPROVED** [review]

**Review by yuce — APPROVED** [review]

**Pull request merged**